src/Repository/RedirectRepository.php

```diff
@@ -7,28 +7,30 @@
 use Setono\SyliusRedirectPlugin\Model\RedirectInterface;
 use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
 
-class RedirectRepository extends EntityRepository
+class RedirectRepository extends EntityRepository implements RedirectRepositoryInterface
 {
     /**
-     * @param string $source
-     *
-     * @return RedirectInterface|null
+     * {@inheritdoc}
      *
      * @throws \Doctrine\ORM\NonUniqueResultException
      */
-    public function findBySource(string $source): ?RedirectInterface
+    public function findEnabledBySource(string $source, bool $onlyNotFound = false): ?RedirectInterface
```

Comment:
Not really since you want to log URL access. The asked URL is not the latest, but the first. We need to have the first, mark it as accessed, then find the last.